packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/javascript/guide.md

```diff
@@ -1,3 +1,406 @@
-## JavaScript
+## About JavaScript and the Okta Sign-In Widget
 
-To come ....
+The Okta Sign-In Widget is a JavaScript library that gives you a fully-featured and customizable sign in experience, which can be used to authenticate users on any website.
+
+Okta uses the Widget as part of its normal sign-in page. If you would like to customize the Widget, then you will need to host it yourself. This guide walks you through the [installation process](#installation) for the Widget, as well as [a few common use cases](#use-cases) for the Widget and how to implement them. The full Widget reference can be found [on GitHub](https://github.com/okta/okta-signin-widget#okta-sign-in-widget).
+
+> A version of the Widget that you can edit in real time can be found here: <https://developer.okta.com/live-widget/>
+
+<img src="/img/okta-sign-in-javascript.png" alt="Screenshot of basic Okta Sign-In Widget" width="400">
+
+## Installation
+
+The first step is to install the Widget. You have two options: linking out to the Okta CDN, or installing locally through `npm` instead.
+
+### CDN
+
+To use the CDN, include this in your HTML:
+
+```html
+<!-- Latest CDN production JavaScript and CSS -->
+<script src="https://global.oktacdn.com/okta-signin-widget/-=OKTA_REPLACE_WITH_WIDGET_VERSION=-/js/okta-sign-in.min.js" type="text/javascript"></script>
+<link href="https://global.oktacdn.com/okta-signin-widget/-=OKTA_REPLACE_WITH_WIDGET_VERSION=-/css/okta-sign-in.min.css" type="text/css" rel="stylesheet"/>
+```
+
+More info, including the latest published version, can be found in the [Widget Documentation](https://github.com/okta/okta-signin-widget#using-the-okta-cdn).
+
+### npm
+
+```bash
+# Run this command in your project root folder.
+npm install @okta/okta-signin-widget@-=OKTA_REPLACE_WITH_WIDGET_VERSION=-
+```
+
+More info, including the latest published version, can be found in the [Widget Documentation](https://github.com/okta/okta-signin-widget#using-the-npm-module).
+
+#### Bundling the Widget
+
+If you are bundling your assets, import them from `@okta/okta-signin-widget`. For example, using [webpack](https://webpack.js.org/):
+
+```javascript
+import OktaSignIn from '@okta/okta-signin-widget';
+import '@okta/okta-signin-widget/dist/css/okta-sign-in.min.css';
+```
+
+> Loading CSS requires the css-loader plugin. You can find more information about it at the [css-loader gitHub repository](https://github.com/webpack-contrib/css-loader#usage).
+
+### Enabling Cross-Origin Access
+
+Because the Widget will be making cross-origin requests, you need to enable Cross Origin Access (CORS) by adding your application's URL to your Okta org's Trusted Origins (in **Security** > **API** > **Trusted Origins**). More information about this can be found on the [Enable CORS](/docs/guides/enable-cors/) page.
```

Comment:
```suggestion
Because the Widget makes cross-origin requests, you need to enable Cross Origin Access (CORS) by adding your application's URL to your Okta org's Trusted Origins (in **Security** > **API** > **Trusted Origins**). See [Enable CORS](/docs/guides/enable-cors/).
```